io/include/pcl/io/pcd_io.h

```diff
@@ -252,12 +252,14 @@ namespace pcl
                             const Eigen::Quaternionf &orientation);
 
       /** \brief Generate the header of a BINARY_COMPRESSED PCD file format
+        * \param[out] os the stream into which to write the header
         * \param[in] cloud the point cloud data message
         * \param[in] origin the sensor acquisition origin
         * \param[in] orientation the sensor acquisition orientation
         */
-      std::string
-      generateHeaderBinaryCompressed (const pcl::PCLPointCloud2 &cloud,
+      int
```

Comment:
I'd also implement this all differently. But you are right, we have to live with the rest of the code, so let's keep new things consistent. All functions here return (-1) for failure and (0) for success, let's keep it this way. But a note in docstring about this will still be helpful.